lib/ElementHandle.js

```diff
@@ -137,24 +151,72 @@ class ElementHandle extends JSHandle {
     return {x, y, width, height};
   }
 
+  /**
+   * @return {!Promise<?{x: number, y: number, width: number, height: number}>}
+   */
+  async _assertBoundingBox() {
+    const boundingBox = await this.boundingBox();
+    if (!boundingBox)
+      throw new Error('Node is not visible');
+    return boundingBox;
+  }
+
   /**
    *
    * @param {!Object=} options
    * @returns {!Promise<Object>}
    */
   async screenshot(options = {}) {
-    await this._scrollIntoViewIfNeeded();
+    let needsViewportReset = false;
+
+    let boundingBox = await this._assertBoundingBox();
+
+    const viewport = Object.assign({}, this._page.viewport());
+
+    const newViewport = {
+      width: Math.max(viewport.width, Math.ceil(boundingBox.width)),
+      height: Math.max(viewport.height, Math.ceil(boundingBox.height)),
+    };
+
+    const initialScroll = await this.executionContext().evaluate(function() {
+      return { scrollX: window.scrollX, scrollY: window.scrollY };
+    });
```

Comment:
So we can reset the scroll; fewer side effects from running the screenshot function.